### PR TITLE
ESLint

### PR DESCRIPTION
**Pull request by robertjd**
'sup ya'll! @rdegges @timothyej @typerandom 

This adds the `.eslintrc` file to the repo, and ports all the JS Hint settings that I've been using across all of our JS projects.  This is a good, sane place to start IMO.  The library code has been patched to meet the linter rules.

Please pull this branch and try running the tests, as they are now configured to run eslint first.

### CODE REVIEW

**Comment by typerandom** [comment]
Awesome, going to review this today. :beers:

**Comment by typerandom** [comment]
Works awesome. I added a npm script command so that in order to just do lint one can execute `$ npm run-script lint`. Also added some more rules to the mix (in line with discussions we've had before). But there are some linting errors that needs to be cleaned up. I'll fix this ASAP.

File: lib/oauth/common.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ module.exports = {
     var oauthQueryStateToken = req.query.state;
     var oauthCookieStateToken = req.cookies.oauthStateToken;
 
-    if (!oauthQueryStateToken || !oauthCookieStateToken) {
+    if (!oauthQueryStateToken || (!oauthCookieStateToken)) {
```
Comment (timothyej) [inline lib/oauth/common.js]: Why the extra parentheses on the second expression?

File: test/handlers/test-post-login-handler.js [hunk 2]
```diff
@@ -28,29 +28,29 @@ function preparePostLoginExpansionTestFixture(stormpathApplication,cb){
...
   });
 
-  var fixture = {
-    expressApp: app,
-    sideEffectData: sideEffectData,
-    sideEffect: null
-  };
-
-  app.on('stormpath.ready', cb.bind(null,fixture));
+  fixture.expressApp.on('stormpath.ready', cb.bind(null,fixture));
```
Comment (timothyej) [inline test/handlers/test-post-login-handler.js]: Forgot space before `fixture`?

File: test/handlers/test-post-registration-handler.js [hunk 3]
```diff
@@ -6,10 +6,8 @@ var uuid = require('uuid');
 
 var helpers = require('../helpers');
 
-function preparePostRegistrationExpansionTestFixture(stormpathApplication,cb){
-
+function preparePostRegistrationExpansionTestFixture(stormpathApplication,cb) {
```
Comment (timothyej) [inline test/handlers/test-post-registration-handler.js]: Forgot space before `cb`?

File: test/handlers/test-post-registration-handler.js [hunk 4]
```diff
@@ -18,7 +16,7 @@ function preparePostRegistrationExpansionTestFixture(stormpathApplication,cb){
     expand: {
       customData: true
     },
-    postRegistrationHandler: function(account,req,res){
+    postRegistrationHandler: function(account,req,res) {
```
Comment (timothyej) [inline test/handlers/test-post-registration-handler.js]: Even more lack of spaces here in arguments, maybe it's not an eslint rule?

**Comment by timothyej** [comment]
@rdegges Other than my minor comments, well done! 

@typerandom Nice! FYI, you can also use `$ npm run lint` instead of `run-script` :) And also, I would prefer if we could use [imperative style when writing commit messages](http://stackoverflow.com/questions/3580013/should-i-use-past-or-present-tense-in-git-commit-messages).

File: test/handlers/test-post-login-handler.js (hunk 2 above)
Comment (typerandom) [inline test/handlers/test-post-login-handler.js]: That is not yet a rule enforced. I add rules and fix things as I go.

File: test/handlers/test-post-registration-handler.js (hunk 3 above)
Comment (typerandom) [inline test/handlers/test-post-registration-handler.js]: As said before, not yet a rule enforced.

File: test/handlers/test-post-registration-handler.js (hunk 4 above)
Comment (typerandom) [inline test/handlers/test-post-registration-handler.js]: Yep, not yet a rule. But will have to add that. Still many rules to add :)

File: lib/oauth/common.js (hunk 1 above)
Comment (typerandom) [inline lib/oauth/common.js]: Strange. I didn't add that on purpose.

**Comment by typerandom** [comment]
@timothyej Nice, didn't know about the `run-script` alias :)

File: lib/oauth/common.js (hunk 1 above)
Comment (timothyej) [inline lib/oauth/common.js]: No, @rdegges did :)

File: test/handlers/test-post-registration-handler.js (hunk 4 above)
Comment (timothyej) [inline test/handlers/test-post-registration-handler.js]: :+1:

**Comment by timothyej** [comment]
When this is merged, I will extract it into its own repository so it can be shared among our other projects. Like Walmart has done: <https://github.com/walmartlabs/eslint-config-defaults>

**Comment by typerandom** [comment]
Yeah, that I think was something we all agreed on :) :+1:

**Comment by timothyej** [comment]
Yep, just posted it here as a reminder, or else I will forget ;)

**Comment by timothyej** [comment]
I'm merging this now, new rules and changes should be issued in separate PRs.

**Pull request merged**